doc_comments_ai/app.py

```diff
@@ -101,52 +104,64 @@ def run():
-                    f"Consider using {utils.get_bold_text('--gpt4')} "
-                    f"or {utils.get_bold_text('--gpt3_5-16k')}. "
-                    "Skipping for now..."
-                )
+        if args.guided:
+            print(f"Generate doc for {utils.get_bold_text(method_name)}? (y/n)")
+            if not input().lower() == "y":
                 continue
 
-            spinner = yaspin(text=f"🔧 Generating doc comment for {method_name}...")
-            spinner.start()
+        method_source_code = node.method_source_code
 
-            documented_method_source_code = llm_wrapper.generate_doc_comment(
-                programming_language.value, method_source_code, args.inline
+        tokens = utils.count_tokens(method_source_code)
+        total_original_tokens += tokens
+        if tokens > 2048 and not (args.gpt4 or args.gpt3_5_16k):
+            print(
+                f"⚠️  Method {method_name} has too many tokens. "
+                f"Consider using {utils.get_bold_text('--gpt4')} "
+                f"or {utils.get_bold_text('--gpt3_5-16k')}. "
+                "Skipping for now..."
             )
+            continue
 
-            generated_doc_comments[
-                method_source_code
-            ] = utils.extract_content_from_markdown_code_block(
-                documented_method_source_code
-            )
+        spinner = yaspin(text=f"🔧 Generating doc comment for {method_name}...")
+        spinner.start()
 
-            spinner.stop()
+        doc_comment_result = llm_wrapper.generate_doc_comment(
+            programming_language.value, method_source_code, args.inline, args.comment_with_source_code
+        )
 
-            print(f"✅ Doc comment for {method_name} generated.")
+        generated_tokens = utils.count_tokens(doc_comment_result)
+        total_generated_tokens += generated_tokens
 
-    file.close()
+        if args.inline or args.comment_with_source_code:
+            parsed_doc_comment = utils.extract_content_from_markdown_code_block(
+                doc_comment_result
+            )
+            utils.write_code_snippet_to_file(
+                file_name, method_source_code, parsed_doc_comment
+            )
+        else:
+            parsed_doc_comment = utils.extract_comments_from_markdown_code_block(
+                programming_language.value, doc_comment_result
+            )
+            utils.write_only_comments_to_file(
+                file_name, method_source_code, parsed_doc_comment
+            )
 
-    for original_code, generated_doc_comment in generated_doc_comments.items():
-        utils.write_code_snippet_to_file(
-            file_name, original_code, generated_doc_comment
-        )
+        spinner.stop()
+
+        print(f"✅ Doc comment for {method_name} generated.")
+    
+    print(f"📊 Total Input Tokens: {total_original_tokens}")
+    print(f"🚀 Total Generated Tokens: {total_generated_tokens}")
```

Comment:
I like these outputs! Maybe we can add a `--verbose` argument and only output token details in verbose mode.

Reply:
Sure nice idea! I'll make this change.